packages/ckeditor5-emoji/src/emojimention.ts

```diff
@@ -0,0 +1,252 @@
+	 */
+	declare private _emojiDatabasePlugin: EmojiDatabase;
+
+	/**
+	 * Defines a number of displayed items in the auto complete dropdown.
+	 *
+	 * It includes the "Show all emoji..." option if the `EmojiPicker` plugin is loaded.
+	 */
+	private readonly _emojiDropdownLimit: number;
+
+	/**
+	 * @inheritDoc
+	 */
+	public static get requires() {
+		return [ EmojiDatabase, 'Mention' ] as const;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static get pluginName() {
+		return 'EmojiMention' as const;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static override get isOfficialPlugin(): true {
+		return true;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor( editor: Editor ) {
+		super( editor );
+
+		this.editor.config.define( 'emoji', {
+			dropdownLimit: 6
+		} );
+
+		this._emojiDropdownLimit = editor.config.get( 'emoji.dropdownLimit' )!;
+
+		const mentionFeedsConfigs = editor.config.get( 'mention.feeds' )! as Array<MentionFeed>;
+		const mergeFieldsPrefix = editor.config.get( 'mergeFields.prefix' )! as string;
+		const markerAlreadyUsed = mentionFeedsConfigs.some( config => config.marker === EMOJI_MENTION_MARKER );
+		const isMarkerUsedByMergeFields = mergeFieldsPrefix ? mergeFieldsPrefix[ 0 ] === EMOJI_MENTION_MARKER : false;
+
+		if ( markerAlreadyUsed || isMarkerUsedByMergeFields ) {
+			/**
+			 * The `marker` in the `emoji` config is already used by other plugin configuration.
+			 *
+			 * @error emoji-config-marker-already-used
+			 * @param {string} marker Used marker.
+			 */
+			logWarning( 'emoji-config-marker-already-used', { marker: EMOJI_MENTION_MARKER } );
+
+			return;
+		}
+
+		this._setupMentionConfiguration( mentionFeedsConfigs );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public init(): void {
+		const editor = this.editor;
+
+		this._emojiPickerPlugin = editor.plugins.has( 'EmojiPicker' ) ? editor.plugins.get( 'EmojiPicker' ) : null;
+		this._emojiDatabasePlugin = editor.plugins.get( 'EmojiDatabase' );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public afterInit(): void {
+		const editor = this.editor;
+
+		// Skip overriding the `mention` command listener if the emoji database is not loaded.
+		if ( !this._emojiDatabasePlugin.isDatabaseLoaded() ) {
+			return;
+		}
+
+		editor.once( 'ready', this._overrideMentionExecuteListener.bind( this ) );
+	}
+
+	/**
+	 * Initializes the configuration for emojis in the mention feature.
+	 */
+	private _setupMentionConfiguration( mentionFeedsConfigs: Array<MentionFeed> ): void {
+		const emojiMentionFeedConfig = {
+			marker: EMOJI_MENTION_MARKER,
+			dropdownLimit: this._emojiDropdownLimit,
+			itemRenderer: this._customItemRendererFactory( this.editor.t ),
+			feed: this._queryEmojiCallbackFactory()
+		};
+
+		this.editor.config.set( 'mention.feeds', [ ...mentionFeedsConfigs, emojiMentionFeedConfig ] );
+	}
+
+	/**
+	 * Returns the `itemRenderer()` callback for mention config.
+	 */
+	private _customItemRendererFactory( t: LocaleTranslate ) {
```

Comment:
Missing the returned type. Shouldn't it be [`ItemRenderer`](https://ckeditor.com/docs/ckeditor5/latest/api/module_mention_mentionconfig-ItemRenderer.html)?

Reply:
Done in b53e60f9c094d5f821e1ee08726eb12038509a25.